### ISSUE TITLE
DDLDatabase fails with MySQL UNIQUE INDEX clause

### ISSUE THREAD

**Issue by anuraaga** [jOOQ/jOOQ#7022]
JOOQ 3.10.3

```
org.jooq.impl.ParserException: C42000_NO_SUBCLASS: Expected tokens: []: TAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
  UNIQUE [*]INDEX `id_epoch` (id, epoch)
);

CREATE TABLE template
        at org.jooq.impl.ParserImpl$ParserContext.unexpectedToken(ParserImpl.java:5791)
        at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:547)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:345)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:337)
        at org.jooq.util.ddl.DDLDatabase.create0(DDLDatabase.java:124)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:237)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:226)
        at org.jooq.util.AbstractDatabase.getDialect(AbstractDatabase.java:208)
        at org.jooq.util.AbstractGenerator.logDatabaseParameters(AbstractGenerator.java:131)
        at org.jooq.util.JavaGenerator.generate(JavaGenerator.java:271)
        at org.jooq.util.GenerationTool.run(GenerationTool.java:642)
        at org.jooq.util.GenerationTool.generate(GenerationTool.java:203)
        at org.jooq.util.GenerationTool.main(GenerationTool.java:175)
```

**Comment by lukaseder** [comment]
Thank you very much for reporting. Indeed, we should parse and ignore the optional `INDEX` and `KEY` keywords after `UNIQUE`. Do note that you can remove the keyword on your side as a workaround.

**Comment by lukaseder** [comment]
Implemented in 3.11.0. To be merged to 3.10.4 (#7031)
